src/libs/ReportUtils.js

```diff
@@ -1933,7 +1973,11 @@ function buildOptimisticExpenseReport(chatReportID, policyID, payeeAccountID, to
  * @returns {Array}
  */
 function getIOUReportActionMessage(iouReportID, type, total, comment, currency, paymentType = '', isSettlingUp = false) {
-    let amount = CurrencyUtils.convertToDisplayString(total, currency);
+    const amount =
+        type === CONST.IOU.REPORT_ACTION_TYPE.PAY
+            ? CurrencyUtils.convertToDisplayString(getMoneyRequestTotal(getReport(iouReportID)), currency)
```

Comment:
When we create all optimistic reports, report actions etc the report is not readily available in Onyx and even though this function gets `total` as a param the total is shown as zero for the first time the optimistic actions were created. This caused #43907